src/search/merge_and_shrink/merge_and_shrink_algorithm.cc

```diff
@@ -55,8 +56,65 @@ MergeAndShrinkAlgorithm::MergeAndShrinkAlgorithm(
       log(utils::get_log_for_verbosity(verbosity)),
       main_loop_max_time(main_loop_max_time),
       starting_peak_memory(0) {
+    handle_shrink_limit_defaults();
+}
+
+void MergeAndShrinkAlgorithm::handle_shrink_limit_defaults() {
+    // If none of the two state limits has been set: set default limit.
+    if (max_states == -1 && max_states_before_merge == -1) {
+        max_states = 50000;
+    }
+
+    // If one of the max_states options has not been set, set the other
+    // so that it imposes no further limits.
+    if (max_states_before_merge == -1) {
+        max_states_before_merge = max_states;
+    } else if (max_states == -1) {
+        if (utils::is_product_within_limit(
+                max_states_before_merge, max_states_before_merge, INF)) {
+            max_states = max_states_before_merge * max_states_before_merge;
+        } else {
+            max_states = INF;
+        }
+    }
+
+    if (max_states_before_merge > max_states) {
+        max_states_before_merge = max_states;
+        if (log.is_warning()) {
+            log << "WARNING: "
+                << "max_states_before_merge exceeds max_states, "
+                << "correcting max_states_before_merge." << endl;
+        }
+    }
+
+    utils::verify_comparison(max_states, 1, greater_equal<>(),
+                             "Transition system size must be at least 1.");
+
+    utils::verify_comparison(max_states_before_merge, 1, greater_equal<>(),
+                             "Transition system size before merge must be at least 1.");
+
+    if (shrink_threshold_before_merge == -1) {
+        shrink_threshold_before_merge = max_states;
+    }
+
+    utils::verify_comparison(shrink_threshold_before_merge, 1, greater_equal<>(),
+                             "Threshold must be at least 1.");
+
+
+    if (shrink_threshold_before_merge > max_states) {
+        shrink_threshold_before_merge = max_states;
+        if (log.is_warning()) {
+            log << "WARNING: "
+                << "threshold exceeds max_states, "
+                << "correcting threshold." << endl;
+        }
+    }
+
+    // TODO are these asserts needed?
     assert(max_states_before_merge > 0);
```

Comment:
It is strange to write it as `> 0` here and `>= 1` in the test above. For the same reason, I would also write teh assumptions in lines 114, 115, and 118 in the same direction, rather than using `a < b` once and `c > b` another time.

Also, the assert just after the test seems redundant. Maybe the asserts should be in the constructor to show readers at one glance what properties we guarantee after construction.